### PR TITLE
[TRA-102] Add API to get asset positions for parent subaccount

### PR DESCRIPTION
**Pull request by shrenujb**
### Changelist
Add API to get asset positions for parent subaccount

### Test Plan
Added unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/comlink/__tests__/controllers/api/v4/asset-positions-controller.test.ts [hunk 1]
```diff
@@ -148,18 +150,164 @@ describe('asset-positions-controller#V4', () => {
       const response: request.Response = await sendRequest({
         type: RequestMethod.GET,
         path: `/v4/assetPositions?address=${testConstants.defaultAddress}` +
-          `&subaccountNumber=${testConstants.defaultSubaccount.subaccountNumber}`,
+            `&subaccountNumber=${testConstants.defaultSubaccount.subaccountNumber}`,
```
Comment (Christopher-Li) [inline indexer/services/comlink/__tests__/controllers/api/v4/asset-positions-controller.test.ts:153]: nit: why'd you change the indentation on all of these lines above?

Comment (shrenujb) [reply indexer/services/comlink/__tests__/controllers/api/v4/asset-positions-controller.test.ts:153]: not intentional. ide seems to have a mind of of its own
You can look at the changes without whitespace and it should make it easier to review

File: indexer/services/comlink/src/request-helpers/request-transformer.ts [hunk 2]
```diff
@@ -127,6 +127,21 @@ export function assetPositionToResponseObject(
   };
 }
 
+export function assetPositionMultipleSubaccountsToResponseObject(
```
Comment (vincentwschau) [inline indexer/services/comlink/src/request-helpers/request-transformer.ts:130]: nit: Maybe add a description?

File: indexer/services/comlink/__tests__/controllers/api/v4/asset-positions-controller.test.ts [hunk 3]
```diff
@@ -148,18 +150,164 @@ describe('asset-positions-controller#V4', () => {
...
       });
 
       expect(response.body.positions).toEqual(
         [{
           symbol: testConstants.defaultAsset.symbol,
+          // funding index difference = 10050 (height 3) - 10000 (height 0) = 50
+          // size = 10000 (initial size) - 50 (funding index diff) * 10(position size)
```
Comment (vincentwschau) [inline indexer/services/comlink/__tests__/controllers/api/v4/asset-positions-controller.test.ts:160]: 🙏 Thanks for adding the explanation.

**Review by vincentwschau — APPROVED** [review]

File: indexer/services/comlink/src/controllers/api/v4/asset-positions-controller.ts [hunk 4]
```diff
@@ -106,47 +115,159 @@ class AssetPositionsController extends Controller {
...
+// Helper function to adjust the asset positions with the unsettled funding
+// per subaccount
+async function adjustAssetPositionsWithFunding(
+  assetPositions: AssetPositionFromDatabase[],
+  perpetualPositions: PerpetualPositionFromDatabase[],
+  idToAsset: AssetById,
+  subaccount: SubaccountFromDatabase,
+  latestBlock: BlockFromDatabase,
+): Promise<AssetPositionsMap> {
+  let assetPositionsMap: AssetPositionsMap = _.chain(assetPositions)
```
[the remark this answers is not in the document]
Comment (shrenujb) [reply indexer/services/comlink/src/controllers/api/v4/asset-positions-controller.ts:232]: It is being updated towards the end of the func

File: indexer/services/comlink/src/request-helpers/request-transformer.ts (hunk 2 above)
Comment (shrenujb) [reply indexer/services/comlink/src/request-helpers/request-transformer.ts:130]: Removed unused func

File: indexer/services/comlink/src/controllers/api/v4/asset-positions-controller.ts [hunk 5]
```diff
@@ -106,47 +115,159 @@ class AssetPositionsController extends Controller {
...
+      perpetualPositions,
+      'subaccountId',
+    );
+
+    // For each subaccount, adjust the asset positions with the unsettled funding and return the
+    // asset positions per subaccount
+    let assetPositionsResponse: AssetPositionResponseObject[] = [];
+    for (const subaccount of subaccounts) {
+      const adjustedAssetPositionsMapForSubaccount:
+      AssetPositionsMap = await adjustAssetPositionsWithFunding(
```
[the remark this answers is not in the document]
Comment (shrenujb) [reply indexer/services/comlink/src/controllers/api/v4/asset-positions-controller.ts]: Good call. Done

**Pull request merged**